app/views/shopping_list_item_statuses/deactivate.js.erb

```diff
@@ -1,8 +1,11 @@
+// Remove item
+document.getElementById("<%= dom_id(@shopping_list_item) %>").remove();
+
+// Hide aisle if this was the last item in it
 if (<%= @remove_aisle %>) {
-  document.getElementById("<%= dom_id(@shopping_list_item.aisle) %>").remove();
-} else {
-  document.getElementById("<%= dom_id(@shopping_list_item) %>").remove();
+  document.getElementById("<%= dom_id(@shopping_list_item.aisle) %>").classList.add('hidden');
```

Comment:
@mikevallano you were on to something with the hide/reveal technique. It fixed the bug we had in prod and also lets us gracefully reinsert items back into their aisles.